tests/unit/internal-session-test.js

```diff
@@ -189,6 +196,38 @@ describe('InternalSession', () => {
     describe('when the restored data does not contain an authenticator factory', () => {
       itDoesNotRestore();
     });
+
+    describe('with older synchronous stores (< v1.1.0)', function() {
+      describe('when restoring the session', function() {
+        describe('when the store resolves restoration', function() {
+          beforeEach(() => {
+            sinon.stub(store, 'restore').returns({ authenticated: { authenticator: 'authenticator' } });
+            sinon.stub(store, 'persist').returns();
+            sinon.stub(store, 'clear').returns();
+          });
+
+          it('is authenticated', () => {
+            return session.restore().then(() => {
+              expect(session.get('isAuthenticated')).to.be.true;
+            });
+          });
+        });
+
+        describe('when the store rejects restoration', function() {
```

Comment:
One option could be to let the store clear it's contents but keep the data in memory to invalidate the session. If invalidation is rejected, the contents should be persisted back into the store again.